tests/lib/rules/prefer-const.js

```diff
@@ -56,6 +56,12 @@ ruleTester.run("prefer-const", rule, {
         "let a; function foo() { if (a) {} a = bar(); }",
         "let a; function foo() { a = a || bar(); baz(a); }",
         "let a; function foo() { bar(++a); }",
+        "let predicate; [typeNode.returnType, predicate] = foo();",
+        "let predicate; let rest; [typeNode.returnType, predicate, ...rest] = foo();",
+        {
+            code: "let predicate; [typeNode.returnType, predicate] = foo();",
+            options: [{ destructuring: "all" }]
+        },
```

Comment:
Can you add a test where the destructuring uses an `ObjectPattern` rather than an `ArrayPattern`? From looking at the code, I'm not sure this case is handled correctly.

```js
let foo;

({ x: bar.baz, y: foo } = qux);
```